zeromq 2017 support

Current it is not possible to install zeromq using vcpkg for Visual Studi 2017

`vcpkg install zeromq:x64-windows`  yields following error:
```


Microsoft (R) Build Engine version 15.1.548.43366
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 10.03.2017 00.31.03.
     1>Project "C:\vcpkg\buildtrees\zeromq\src\zeromq-4.2.0\builds\msvc\vs2015\libzmq\libzmq.vcxproj" on node 1 (default targets).
     1>C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.targets(57,5): error MSB8020: The build tools for v140 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install v140 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\vcpkg\buildtrees\zeromq\src\zeromq-4.2.0\builds\msvc\vs2015\libzmq\libzmq.vcxproj]
     1>Done Building Project "C:\vcpkg\buildtrees\zeromq\src\zeromq-4.2.0\builds\msvc\vs2015\libzmq\libzmq.vcxproj" (default targets) -- FAILED.

Build FAILED.

       "C:\vcpkg\buildtrees\zeromq\src\zeromq-4.2.0\builds\msvc\vs2015\libzmq\libzmq.vcxproj" (default target) (1) ->
       (PlatformPrepareForBuild target) -> 
         C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\Common7\IDE\VC\VCTargets\Microsoft.Cpp.Platform.targets(57,5): error MSB8020: The build tools for v140 (Platform Toolset = 'v140') cannot be found. To build using the v140 build tools, please install v140 build tools.  Alternatively, you may upgrade to the current Visual Studio tools by selecting the Project menu or right-click the solution, and then selecting "Retarget solution". [C:\vcpkg\buildtrees\zeromq\src\zeromq-4.2.0\builds\msvc\vs2015\libzmq\libzmq.vcxproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.69

```

Any way of fixing this?

--- a/ports/zeromq/CONTROL
+++ b/ports/zeromq/CONTROL
@@ -1,3 +1,3 @@
 Source: zeromq
-Version: 4.2.0
+Version: 4.2.2
 Description: The ZeroMQ lightweight messaging kernel is a library which extends the standard socket interfaces with features traditionally provided by specialised messaging middleware products

--- a/ports/zeromq/portfile.cmake
+++ b/ports/zeromq/portfile.cmake
@@ -1,9 +1,12 @@
+set(ZEROMQ_VERSION 4.2.2)
+set(ZEROMQ_HASH 4069813374d4e8d4c0f8debbe85472d0bd24cf644fb1bce748920eadffb81c429d28f523ef424df84fcaa7082b984fab8da57192802585811d37cff066f4e40c)
+
 include(vcpkg_common_functions)
-set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/zeromq-4.2.0)
+set(SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src/libzmq-${ZEROMQ_VERSION})
 vcpkg_download_distfile(ARCHIVE
-    URLS "https://github.com/zeromq/libzmq/releases/download/v4.2.0/zeromq-4.2.0.tar.gz"
-    FILENAME "zeromq-4.2.0.tar.gz"
-    SHA512 3b6f0a1869fb1663ea40b3c3aa088b81399a35c051e4ade2b30bbac60bfceefe6b4403248a4635fb31d33767c1e478342f61c47b0ffdb4501419c13590ebeb96
+    URLS "https://github.com/zeromq/libzmq/archive/v${ZEROMQ_VERSION}.tar.gz"
+    FILENAME "libzmq-${ZEROMQ_VERSION}.tar.gz"
+    SHA512 ${ZEROMQ_HASH}
 )
 vcpkg_extract_source_archive(${ARCHIVE})
 
@@ -14,42 +17,48 @@ else ()
     set(MSBUILD_PLATFORM ${TRIPLET_SYSTEM_ARCH})
 endif()
 
+if(VCPKG_PLATFORM_TOOLSET MATCHES "v141")
+  set(MSVS_VERSION 2017)
+else()
+  set(MSVS_VERSION 2015)
+endif()
+
 if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
     vcpkg_build_msbuild(
-        PROJECT_PATH ${SOURCE_PATH}/builds/msvc/vs2015/libzmq/libzmq.vcxproj
+        PROJECT_PATH ${SOURCE_PATH}/builds/msvc/vs${MSVS_VERSION}/libzmq/libzmq.vcxproj
         RELEASE_CONFIGURATION ReleaseDLL
         DEBUG_CONFIGURATION DebugDLL
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/v140/dynamic/libzmq.dll
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/${VCPKG_PLATFORM_TOOLSET}/dynamic/libzmq.dll
         DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/v140/dynamic/libzmq.dll
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/${VCPKG_PLATFORM_TOOLSET}/dynamic/libzmq.dll
         DESTINATION ${CURRENT_PACKAGES_DIR}/bin
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/v140/dynamic/libzmq.lib
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/${VCPKG_PLATFORM_TOOLSET}/dynamic/libzmq.lib
         DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/v140/dynamic/libzmq.lib
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/${VCPKG_PLATFORM_TOOLSET}/dynamic/libzmq.lib
         DESTINATION ${CURRENT_PACKAGES_DIR}/lib
     )
     vcpkg_copy_pdbs()
 
 else()
     vcpkg_build_msbuild(
-        PROJECT_PATH ${SOURCE_PATH}/builds/msvc/vs2015/libzmq/libzmq.vcxproj
+        PROJECT_PATH ${SOURCE_PATH}/builds/msvc/vs${MSVS_VERSION}/libzmq/libzmq.vcxproj
         RELEASE_CONFIGURATION ReleaseLIB
         DEBUG_CONFIGURATION DebugLIB
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/v140/static/libzmq.lib
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Debug/${VCPKG_PLATFORM_TOOLSET}/static/libzmq.lib
         DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
     )
     file(INSTALL
-        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/v140/static/libzmq.lib
+        ${SOURCE_PATH}/bin/${MSBUILD_PLATFORM}/Release/${VCPKG_PLATFORM_TOOLSET}/static/libzmq.lib
         DESTINATION ${CURRENT_PACKAGES_DIR}/lib
     )
 endif()

